infrastructure/src/main/java/org/corfudb/infrastructure/DynamicTriggerPolicy.java

```diff
@@ -54,8 +66,8 @@ private boolean shouldForceTrigger(CorfuStore corfuStore) {
     @Override
     public boolean shouldTrigger(long interval, CorfuStore corfuStore) {
         DistributedCheckpointerHelper distributedCheckpointerHelper = new DistributedCheckpointerHelper(corfuStore);
-        if (distributedCheckpointerHelper.isCheckpointFrozen()) {
-            log.warn("Compaction has been frozen");
+        if (isCompactionDisabled(corfuStore) || distributedCheckpointerHelper.isCheckpointFrozen()) {
+            log.warn("Compaction has been disabled/frozen");
```

Comment:
For debugging purpose it would be helpful if the log message can differentiate between freeze and disable.